src/core_plugins/vega/index.js

```diff
@@ -0,0 +1,15 @@
+export default kibana => new kibana.Plugin({
+  id: 'vega',
+  require: ['elasticsearch'],
+
+  uiExports: {
+    visTypes: ['plugins/vega/vega_type'],
+    injectDefaultVars: server => ({ vegaConfig: server.config().get('vega') }),
+  },
+
+  config: (Joi) => Joi.object({
+    enabled: Joi.boolean().default(true),
+    enableExternalUrls: Joi.boolean().default(true)
```

Comment:
I am not sure if we want to have this enabled by default and would like @epixa approval for this. Explanation:

If `enableExternalUrls` is set to `true`, you can put URLs into the vega spec, and vega will load (client-side) that URL. Since you can save this visualization, it means that any user who is allowed to create and save visualizations/dashboards, can do this and place it on a dashboard, and that will cause the browser of each (other) visiting user to load that URL in their browser.

In the best case the server to which that URL points, can be used to track the users. I don't know in what way Vega internally handles all that loading and parsing of that URL, but in the worst case the data delivered under this URL could be use to "break" Kibana in another users browser (e.g. maybe the server could - also depending on which user visits - deliver overly large files, that will cause the browser to freeze).

I think, that we should have this set to `false` by default, and require the admin of the instance (i.e. someone with write access to the `kibana.yml`) to enable this manually.

@epixa could you share you point of view?

Reply:
We don't consider linking to external URLs to be a security issue, and you can already do so with URL field formatters, but it is a delicate issue. For now, it's important that all external links use target=_blank, and it's also important that URLs be validated to begin with a valid protocol for external links or have the current install's protocol added for relative links to help prevent embedded JS via anchor tags.  See https://github.com/elastic/kibana/pull/14613 and https://github.com/elastic/kibana/pull/14994 for recent examples.

Reply:
To clarify, I'm assuming this configuration simply allows for external links to be set in `a` tags in the browser and that it has no other special meaning.  Please correct me if I am wrong.

Reply:
To clarify: the external URLs are not just links. An AJAX request will be done towards this URL in the browser of any person watching this visualization.

Reply:
The content in these external urls can be used as input to the vega vis, but I think this is fine. People can also use the url formatter to load urls automatically on page load with an url field format of type image, so I don't think this is much different. At least the option to disable this feature is available here.

@nyurik is it possible to do anything other than a get request? What kind of options can users set? Can custom headers be assigned?

Reply:
@spalger in vanilla Vega, `"url": "..."` is a simple get XHR request (in case of data). Vega also supports images, e.g. `<img href=...>`, in which case the browser does the loading.

In our case, I overload the url sanitization method to throw an error in case enableExternalUrls is false for both of these cases.  For Wikipedia, I had a custom loader that also did domain whitelisting, and adds some extra headers internal to wikipedia (forces the GET request to loose user context). @timroes suggested I add domain whitelister here as well.

There is also a syntax sugar to construct those urls --  (  `"url": {"%type%": "url", "query": {...}}}` ) is automatically converted into a GET request URL using $.params() call.  This just makes writing URLs easier in case parameter values need to be URLencoded, but doesn't avoid security checks.